core/foundation/inc/TClassEdit.h

```diff
@@ -153,6 +154,33 @@ namespace TClassEdit {
       TSplitType &operator=(const TSplitType &) = delete;
    };
 
+   /// A simple RAII helper to remove and readd enclosing _Atomic()
+   /// It expects no spaces at the beginning or end of the string
+   class AtomicTypeNameHandlerRAII {
+      std::string &fTypeName;
+      bool fIsAtomic = false;
+      const char *fgPrefix = "_Atomic(";
+
+   public:
+      AtomicTypeNameHandlerRAII(std::string &typeName) : fTypeName(typeName)
+      {
+         if (0 == fTypeName.find(fgPrefix)) {
+            fIsAtomic = true;
+            fTypeName.erase(0, 8);
+            if (0 == fTypeName.size() || ')' != fTypeName[fTypeName.size() - 1]) {
+               throw std::runtime_error("Cannot remove substring \")\" at the end of typename \""+typeName+"\"");
```

Comment:
Note that the calling code is very unlikely to be exception safe.

Reply:
Yes. The exception is expressive enough to explain why ROOT probably crashed though perhaps?

Reply:
It is probably fine in practice but still iffy.  i.e. it is unknown what state catching this exception would leave ROOT in; we should assume 'unusable' and if we do, then this might as well be a call to `Fatal`.